src/ray/object_manager/pull_manager.cc

```diff
@@ -550,6 +594,47 @@ void PullManager::Tick() {
   }
 }
 
+void PullManager::PinNewObjectIfNeeded(const ObjectID &object_id) {
+  bool active = false;
+  {
+    absl::MutexLock lock(&active_objects_mu_);
+    active = active_object_pull_requests_.count(object_id) > 0;
+  }
+  if (active) {
+    if (TryPinObject(object_id)) {
+      RAY_LOG(DEBUG) << "Pinned newly created object " << object_id;
+    } else {
+      RAY_LOG(DEBUG) << "Failed to pin newly created object " << object_id;
+    }
+  }
```

Comment:
I'm not clear on the multithreading, but we do we need to check whether the request is still active after the pinning is done?

Reply:
Mutations are only done with the lock held, so it's not an issue.

Reply:
I meant that we could leak pins if the object gets deactivated right after the lock is released. Isn't this possible?